app/src/main/java/com/example/plantsapp/presentation/ui/tasksfordays/StatisticsAppResolver.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.plantsapp.presentation.ui.tasksfordays
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class StatisticsAppResolver @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+
+    val intent = Intent().apply {
+        component = ComponentName(
+            PACKAGE_NAME,
+            ACTIVITY_CLASS_NAME
+        )
+    }
+
+    fun isAppInstalled(): Boolean {
+        return context.packageManager.resolveActivity(intent,0) != null
```

Comment:
Check in documentation for what these flags are needed , maybe you need to specify some flags